test/v2/services/PersistenceServiceSpec.scala

```diff
@@ -195,7 +188,7 @@ class PersistenceServiceSpec
       summaries =>
         val expected = PaginationMessageSummary(TotalCount(summaries.length), summaries)
         when(
-          mockConnector.getMessages(EORINumber(any()), any(), MovementId(any()), any(), eqTo(Some(PageNumber(2))), eqTo(Some(ItemCount(30))), any())(
+          mockConnector.getMessages(EORINumber(any()), any(), MovementId(any()), any(), eqTo(Some(PageNumber(2))), ItemCount(eqTo(ItemCount(30).value)), any())(
```

Comment:
`ItemCount(eqTo(ItemCount(30).value))` is really redundant -- just needs to be `ItemCount(eqTo(30))` (and the same elsewhere in the tests)

Reply:
I already tried the way you are suggesting but I was getting ```NullPointerException```.Just now also I checked and I am getting ```NullPointerException``` when I am using ```ItemCount(eqTo(30))```